validations.go

```diff
@@ -9,7 +9,7 @@ var reStringID = regexp.MustCompile(`^[a-zA-Z0-9\-\._]+$`)
 
 // validString checks if the given input is present and non-empty.
 func validString(v *string) bool {
-	return v != nil && *v != ""
+	return v != nil
```

Comment:
This is more thinking out loud than an actual review since I'm still learning golang and learning the Terraform provider structure.

I think removing the check for a non-empty string may let the `key` attribute of the resource be set to an empty string.
Instead, should the `validString` function call be removed from the `value` attribute [here](https://github.com/hashicorp/go-tfe/blob/master/variable.go#L128-L134)?

Again this is more for learning purposes rather than actual feedback.

Reply:
Good point! Will reach out.